### PR TITLE
[RFR-23] Move libs api from data-collector 6.9.5

### PR DESCRIPTION
**Pull request by hb0**
You suggested some time ago to move the library which is used by all Vert.X APIs out of data-collector, which makes sense.

I added the library now to this new repository which is public as the data-collector is public, too, and depends on this library.

### CODE REVIEW

**Review by hb0 — COMMENTED** [review]
pre-reviewed

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
+= API Library
+
+image:https://img.shields.io/badge/vert.x-4.3.2-purple.svg[link="https://vertx.io"]
+image:https://github.com/cyface-de/api/workflows/Cyface%20API/badge.svg[link="https://github.com/cyface-de/api/actions"]
+
+This library is used by the Cyface Vert.X APIs like the https:///github.com/cyface-de/data-collector[Cyface Data Collector] software.
+
+It contains share code required by most of the APIs, like authentication and setup.
```
Comment (muthenberg) [inline README.adoc]: shared

**Comment by muthenberg** [comment]
> You suggested some time ago to move the library which is used by all Vert.X APIs out of data-collector, which makes sense.
> 
> I added the library now to this new repository which is public as the data-collector is public, too, and depends on this library.

That is true. However

File: libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java]: This file and multiple ones like it are not required by the Data-Collector. Are they really necessary as part of this API library? I guess mostly the Authentication and Authorization stuff is really shared code. Everything else seems to belong to one of the other projects and should probably be reintegrated there (but I might be wrong I have no total overview at the moment).

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalStrategy.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalStrategy.java]: This is probably no shared code as well.

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithSensorData.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithSensorData.java]: This is probably no shared code.

File: libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithoutSensorData.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetrievalWithoutSensorData.java]: this is probably no shared code

File: libs/api/src/main/java/de/cyface/api/MeasurementRetriever.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/MeasurementRetriever.java]: Also probably no shared code

File: libs/api/src/main/java/de/cyface/api/PauseAndResumeStrategy.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (muthenberg) [inline libs/api/src/main/java/de/cyface/api/PauseAndResumeStrategy.java]: Couldn't we move this back to the data-collector. Is it required anywhere else?

File: README.adoc [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+If you require this software under a closed source license for you own projects, please https://www.cyface.de/#kontakt[contact us].
+
+Changes between versions are found in the link:https://github.com/cyface-de/api/releases[Release Section].
+
+The project uses link:https://gradle.org/[Gradle] as the build system.
+It is set up as a Gradle multi project build.
+There is one general categories of sub-projects.
+The libraries shared by several Cyface executables which are grouped in `libs`.
```
Comment (muthenberg) [inline README.adoc:17]: Since there are no executables here, why bother with an additional nesting? Only makes the project setup more complex.

**Review by muthenberg — CHANGES_REQUESTED** [review]
Some general comments, where I might be wrong, but if I am I would like to get a better understanding of how everything has moved around.

File: libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java (hunk 2 above)
Comment (hb0) [reply libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java]: Right. I moved them to the `data-provider` after merging `data-provider` and `data-exporter` as you recommended a while back (https://cyface.atlassian.net/browse/RFR-24).
Done. `api:1.0.0-alpha2`.

File: libs/api/src/main/java/de/cyface/api/PauseAndResumeStrategy.java (hunk 7 above)
Comment (hb0) [reply libs/api/src/main/java/de/cyface/api/PauseAndResumeStrategy.java]: The `libs/api/Authorizer` has to know if it needs to pause/resume requests, depending on what the API endpoints do (stream data or parse all before continuing).

Thus, it has to stay in `libs/apis`.

File: README.adoc (hunk 8 above)
Comment (hb0) [reply README.adoc:17]: Applied to:
- in `api` repo
- in `data-collector` repo
- in `data-provider` repo

**Review by hb0 — COMMENTED** [review]
pre-reviewed

**Review by muthenberg — APPROVED** [review]

**Pull request merged**